### PR TITLE
Public key qr

### PR DESCRIPTION
**Pull request by Rockindash**
Added QR Code to show vault public keys, Issue #836 and #849

### CODE REVIEW

**Review by flypaper0 — APPROVED** [review]

File: VultisigApp/VultisigApp/Views/Components/VaultDetailQRCode.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                .opacity(0.7)
+        }
+    }
+    
+    func getQRCode(vault: Vault) -> Image {
+        let name = vault.name
+        let ecdsaKey = vault.pubKeyECDSA
+        let eddsaKey = vault.pubKeyEdDSA
+        let hexCode = vault.hexChainCode
+        let id = "sha256(\(name) - \(ecdsaKey) - \(eddsaKey) - \(hexCode))"
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Views/Components/VaultDetailQRCode.swift:101]: @Rockindash , id , I mean we should compute the SHA256 hash of "name-ecdsakey-eddsakey-hexode" , not plain text

**Review by johnnyluo — CHANGES_REQUESTED** [review]
see comments

**Review by enriquesouza — CHANGES_REQUESTED** [review]

File: VultisigApp/VultisigApp/Views/Components/VaultDetailQRCode.swift [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+    
+    func getQRCode(vault: Vault) -> Image {
+        let name = vault.name
+        let ecdsaKey = vault.pubKeyECDSA
+        let eddsaKey = vault.pubKeyEdDSA
+        let hexCode = vault.hexChainCode
+        let id = "\(name)-\(ecdsaKey)-\(eddsaKey)-\(hexCode)".sha256()
+        
+        let data = """
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Views/Components/VaultDetailQRCode.swift:103]: This generate a string , which is not a valid json , can we please define a struct , and use json encode?

**Review by johnnyluo — APPROVED** [review]
LGTM

**Review by enriquesouza — APPROVED** [review]
LGTM

**Pull request merged**